### PR TITLE
Add JS API for compiling-to-JavaScript functionality

### PR DESCRIPTION
**Pull request by cwalther**
This commit splits the compile-to-JavaScript command-line tool _bin.js_ into two files, one that contains the actual functionality and one that implements the command line processing. The functionality is then additionally exposed as a JavaScript API, `require('protocol-buffers').compiletojs()`.

This is useful for programmatic use from Gulp and also makes it possible to run the compiler on multi-file schemas that first need to be resolved using _resolve-protobuf-schema_.

No new tests are necessary since `npm run test-compiled` now automatically takes this code path.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -35,3 +36,7 @@ module.exports = function (proto, opts) {
 
   return new Messages()
 }
+
+module.exports.compiletojs = function (proto) {
```
Comment (mafintosh) [inline index.js]: can we name this just `toJS(proto)` or `compileToJS(proto)` ?

**Comment by mafintosh** [comment]
Cool stuff! Added a minor comment :)

**Comment by cwalther** [comment]
Changed the function name to `protobuf.toJS()`. I left the file name because its exported function was already named `compile()` before.

**Comment by mafintosh** [comment]
awesome stuff

**Comment by mafintosh** [comment]
out in 4.1.0 :tada:

**Comment by mafintosh** [comment]
btw, if you wanna add support for resolving imports in the bin.js compiler i'd be all for it :)

**Comment by mafintosh** [comment]
No rush (and no obligation also :))

resolve-protobuf-schema only has a dep on protocol-buffers-schema so it's very minimal overhead
https://github.com/mafintosh/resolve-protobuf-schema/blob/master/package.json

When compiling to JS it wouldn't be included anyway, which is v neat

**Pull request merged**